### PR TITLE
KAFKA-12630: Remove deprecated KafkaClientSupplier#getAdminClient in Streams

### PR DESCRIPTION
**Pull request by guozhangwang**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by guozhangwang** [comment]
ping @ableegoldman @abbccdda

**Review by ableegoldman — APPROVED** [review]
LGTM, thanks for the PR

File: streams/src/main/java/org/apache/kafka/streams/KafkaClientSupplier.java [hunk 1]
```diff
@@ -31,29 +30,14 @@
...
     /**
      * Create an {@link Admin} which is used for internal topic management.
      *
      * @param config Supplied by the {@link java.util.Properties} given to the {@link KafkaStreams}
      * @return an instance of {@link Admin}
      */
-    @SuppressWarnings("deprecation")
     default Admin getAdmin(final Map<String, Object> config) {
-        return getAdminClient(config);
+        throw new UnsupportedOperationException("Implementations of KafkaClientSupplier should implement the getAdmin() method.");
```
Comment (ijuma) [inline streams/src/main/java/org/apache/kafka/streams/KafkaClientSupplier.java:40]: Would we want to remove this default implementation too? Not saying we should, but what's the benefit of keeping it?

Comment (guozhangwang) [reply streams/src/main/java/org/apache/kafka/streams/KafkaClientSupplier.java:40]: I think we can; I was originally worried if I change it from `default` to a pure interface would that be binary compatible --- did some research and it seems okay.

**Pull request merged**